lib/streamlit/elements/widgets/button.py

```diff
@@ -551,7 +550,17 @@ def _download_button(
         ctx: ScriptRunContext | None = None,
     ) -> bool:
         key = to_key(key)
+
+        # Importing these functions here to avoid circular imports
+        from streamlit.elements.utils import (
+            check_callback_rules,
+            check_session_state_rules,
+            check_widget_usage,
+        )
+
+        check_widget_usage()
         check_session_state_rules(default_value=None, key=key, writes_allowed=False)
+        check_callback_rules(self.dg, on_click)
```

Comment:
Sanity-check: This seems to be a new check that did not exist before. Could you elaborate briefly why this new check is needed for buttons (it looks like you didn't add this check to other elements)

Reply:
My assumption was that this was something that we accidentally missed with the button since all other widgets, including `download_button`, have this check.
Looking closer into this, the reason why it was probably left out was that the only check inside `check_callback_rules` is form related and a normal `button` cannot be used in form. 
But I also think its probably good to keep this in for consistency, e.g. if we add another not-form related check to `check_callback_rules`.